app/managers/performance_metrics_manager/performance_metrics_batcher.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppState, Platform, type AppStateStatus} from 'react-native';
+
+import {sendPerformanceReport} from '@actions/remote/performance';
+import DatabaseManager from '@database/manager';
+import {getConfigValue} from '@queries/servers/system';
+import {toMilliseconds} from '@utils/datetime';
+import {logDebug} from '@utils/log';
+
+const MAX_BATCH_SIZE = 100;
+const INTERVAL_TIME = toMilliseconds({seconds: 60});
+
+class Batcher {
+    private started = false;
+    private batch: PerformanceReportMeasure[] = [];
+    private serverUrl: string;
+    private sendTimeout: NodeJS.Timeout | undefined;
+    private lastAppStateIsActive = AppState.currentState === 'active';
+
+    constructor(serverUrl: string) {
+        this.serverUrl = serverUrl;
+        AppState.addEventListener('change', (appState) => this.onAppStateChange(appState));
+    }
+
+    private start() {
+        this.started = true;
+        clearTimeout(this.sendTimeout);
+        this.sendTimeout = setTimeout(() => this.sendBatch(), INTERVAL_TIME);
+    }
+
+    private async sendBatch() {
+        clearTimeout(this.sendTimeout);
+        if (this.batch.length === 0) {
+            return;
+        }
+
+        const toSend = this.getReport();
+
+        // Empty the batch as soon as possible to avoid race conditions
+        this.batch = [];
+        this.started = false;
```

Comment:
I think this needs to be set to false when we clear the timeout above, or else I think we could end up in a scenario where this is still set to true without a timeout running which would cause `addToBatch` to never start the timeout again.

Related to that, do we actually need to track `started` separately from `sendTimeout`? If we set `sendTimeout` to something `-1` or something similar whenever its cleared, then we could use `sendTimeout !== -1` too determine if the timeout is running instead of needing another variable to do that

Reply:
Re: using two variables
I feel it is more readable to keep the `started` variable. That way each variable has only one purpose. But 0/5

Re: setting started to false when we clear the timeout
I will do it. Nevertheless, that should never happen. `started` only sets to true when we add an element. If we add an element, the batch should not be empty. The same could be said about the timeout, so we could move the timeout further down in the function. 0/5 on this.